game-core/src/main/java/games/strategy/engine/framework/startup/ui/panels/main/game/selector/GameSelectorModel.java

```diff
@@ -186,10 +189,28 @@ public void loadDefaultGameSameThread() {
         // can load. (ie: if a previous version of triplea was using running a game within its
         // root folder, we shouldn't open it)
         .filter(
-            defaultGame ->
-                defaultGame.contains(ClientFileSystemHelper.getUserRootFolder().toURI().toString()))
-        .filter(defaultGame -> new File(defaultGame).exists())
+            defaultGame -> {
+              String rootFolder = ClientFileSystemHelper.getUserRootFolder().toURI().toString();
```

Comment:
There is a good bit going on in this filter method. I worry that this method has a mixture of abstraction levels. From what I've seen, it's generally best to keep most logic out of lambdas and try to convert them to be method references by extracting that logic to a helper method.

WDYT of extracting these filter predicates to helper methods? In that case the filter chain would stay quite high level, eg:

```
ClientSetting.defaultGameUri
   .getValue()
   .filter(GameSelectorModel::defaultGameIsInUserRootFolder)
   .filter(GameSelectorModel::defaultGameExists)
   .map(...)
     
```